src/Exception/DomainException.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Common\Exception;
+
+use DomainException as SplDomainException;
+use InvalidArgumentException;
+use Throwable;
+
+class DomainException extends SplDomainException
+{
+    public const DEFAULT_STATUS_CODE = 500;
+
+    /**
+     * @var string
+     */
+    private $token;
```

Comment:
The token is the core of the exceptions, and having anything change the logic or change the value would break any systems relying upon the token. That is why it is private.